pkg/cmd/samples/create.go

```diff
@@ -130,10 +136,11 @@ To see supported samples, run 'stripe samples list'`, args[0])
 	// directory, the user needs to select which integration they
 	// want to work with (if selectedSamplelicable) and which language they
 	// want to copy
-	err = sample.SelectOptions()
+	selectedConfig, err := PromptSampleConfig(sample)
 	if err != nil {
 		return err
 	}
+	sample.SelectedConfig = *selectedConfig
```

Comment:
I think this function still has a lot of implementation details that should be moved into the samples package.

Things such as checking if files/dir exists, if the sample is already up to date, etc.

Reply:
Agree, a single function that creates the sample and returns (path, error) would be ideal. I think this will make grpc implementation simpler as well.

Reply:
I took a look at this and I think it's in scope because the grpc method will need this logic.

However, a problem with moving out the file existence check, sample is up to date check, etc, is that the command wants to print messages to the terminal while those checks are running. If we move out that logic, we would have to think about where to put the spinner and print statements.

If we want to move the logic to the samples package but keep the spinner and print statements in the cobra command, we could maybe use a channel? The `samples.Create` method could write a status to the channel: Downloading, Copying, Configuring, etc, and the cobra command could read the status and write the spinner and print statements asynchronously. What do you think of this approach?

Reply:
Agreed, I think channels is the way to go here to stream the output.

Reply:
Sounds good I will add this!